### PR TITLE
Add 24bit support

### PR DESCRIPTION
**Pull request by Hyun-je**
- Added support for 24-bit external dependencies.
- Fix endian calculation.

### CODE REVIEW

**Comment by eldarsadykov** [comment]
Hello,
Thank you very much for your contribution!
24 bit definitely works, thank you for that. One problem is that on my system the pull request crashes unless I uncomment this:
```swift
class RNBOAudioEngine {
   // ... 
    private func initInput() {
        let input = self.engine.inputNode
        //let format = input.inputFormat(forBus: 0) // <- uncomment this to avoid crash
        let format = avAudioUnit!.inputFormat(forBus: 0) // <- convert to comment this
        
        if (format.channelCount > 0) {
           engine.connect(input, to: avAudioUnit!, format: format) // <- crash in here: Thread 1: "required condition is false: IsFormatSampleRateAndChannelCountValid(hwFormat)"
        }
    }
```
It is absolutely fine, if we just make this fix and merge, but it would be very good if we understood why this is happening.

Best regards,
Eldar

**Comment by Hyun-je** [comment]
Hello. Thank you for your feedback.

I've tested it under a few conditions, but I can't reproduce the crash you mentioned.

Can you tell me more about the environment you tested in?

**Comment by eldarsadykov** [comment]
Sure, I use macOS Sonoma 14.0, XCode 15 on an M1 Pro MacBook Pro 14. Same thing regardless of building for macOS and iOS.

File: SwiftRNBO_Example_multiplatfrom_SwiftUI/SwiftRNBO_Example_multiplatfrom_SwiftUI/RNBO/Interface/RNBOAudioEngine.swift [hunk 1]
```diff
@@ -16,8 +16,8 @@ class RNBOAudioEngine {
     
     private func initInput() {
         let input = self.engine.inputNode
-        //let format = input.inputFormat(forBus: 0)
-        let format = avAudioUnit!.inputFormat(forBus: 0)
+        let format = input.inputFormat(forBus: 0)
```
Comment (eldarsadykov) [inline SwiftRNBO_Example_multiplatfrom_SwiftUI/SwiftRNBO_Example_multiplatfrom_SwiftUI/RNBO/Interface/RNBOAudioEngine.swift:19]: We had to uncomment this in order a crash not to happen on some systems. The error was about channel count mismatch.

**Review by eldarsadykov — APPROVED** [review]
24 bit works

**Pull request merged**